### DISCUSSION TITLE
Accessing GraphQL query string from DgsDataFetchingEnvironment

### DISCUSSION THREAD

**Discussion by jeffcrow5** [Netflix/dgs-framework#1684] [Q&A]
I'm trying to figure out how to access the string of the GraphQL query sent from the client for logging purposes, but haven't been able to figure it out. I've tried digging deeper into the DgsDataFetchingEnvironment in my resolvers, but haven't had any luck. I also haven't found anything in the documentation to solve this problem.

Is there a way to access the GQL query string from the DFE or otherwise?

**Comment by berngp** [comment]
As an option you can use an `Instrumentation` to do that, that will also allow you to log additional details of the query, such as if the execution was successful or not.
